### ISSUE TITLE
Core Dump just after mining a block

### ISSUE THREAD

**Issue by Georges760** [Sjors/bitcoin#55]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

while using SRI pool+jds+jcc+tproxy from git dcaa92dbed49976a5e174941903cf799c68d8eb6, just after finding the block 38172, got a Core Dump crash

### Expected behaviour

not crashing

### Steps to reproduce

```bash
cd bitcoin-sv2-tp-0.1.4
./bin/bitcoind -testnet4 -sv2 -sv2port=8442 -debug=sv2 -conf=/data/bitcoin-sv2/bitcoin.conf -datadir=/data/bitcoin-sv2
```

with /data/bitcoin-sv2/bitcoin.conf :

```conf
server=1
[main]
[test]
[testnet4]
rpcbind=127.0.0.1
rpcbind=::1
rpcbind=192.168.1.75
rpcallowip=192.168.1.47
rpcpassword=****
rpcuser=****
[signet]
[regtest]
```

### Relevant log output

2024-08-06T00:46:49Z Flushed fee estimates to fee_estimates.dat.
2024-08-06T00:47:41Z Saw new header hash=00000000f7298f2400c3514fa2a68a68a77ead56cb15064bfbd85130998987e0 height=38171
2024-08-06T00:47:41Z Saw new cmpctblock header hash=00000000f7298f2400c3514fa2a68a68a77ead56cb15064bfbd85130998987e0 peer=7
2024-08-06T00:47:41Z UpdateTip: new best=00000000f7298f2400c3514fa2a68a68a77ead56cb15064bfbd85130998987e0 height=38171 version=0x20000000 log2_work=70.407958 tx=747408 date='2024-08-06T01:02:07Z' progress=1.000000 cache=0.6MiB(2642txo)
2024-08-06T00:47:41Z CreateNewBlock(): block weight: 6344 txs: 3 fees: 1384 sigops 410
2024-08-06T00:47:41Z [sv2] Send 0x71 NewTemplate id=1130 to client id=1
2024-08-06T00:47:41Z [sv2] Send 0x72 SetNewPrevHash to client id=1
2024-08-06T00:47:41Z CreateNewBlock(): block weight: 6344 txs: 3 fees: 1384 sigops 410
2024-08-06T00:47:41Z [sv2] Send 0x71 NewTemplate id=1131 to client id=2
2024-08-06T00:47:41Z [sv2] Send 0x72 SetNewPrevHash to client id=2
2024-08-06T00:47:41Z [sv2] Received 0x73 RequestTransactionData from client id=2
2024-08-06T00:47:41Z [sv2] Send 0x74 RequestTransactionData.Success to client id=2
2024-08-06T00:47:59Z New block-relay-only v1 peer connected: version: 70015, blocks=38171, peer=704
2024-08-06T00:48:53Z [sv2] Received 0x76 SubmitSolution from client id=2
2024-08-06T00:48:53Z Saw new header hash=000000000000005aad1da1c2646a65e34c94089d7a51a9fe1ae3118169f18c82 height=38172
2024-08-06T00:48:53Z UpdateTip: new best=000000000000005aad1da1c2646a65e34c94089d7a51a9fe1ae3118169f18c82 height=38172 version=0x26b4c000 log2_work=70.408134 tx=747412 date='2024-08-06T00:47:41Z' progress=1.000000 cache=0.6MiB(2644txo)
2024-08-06T00:48:53Z CreateNewBlock(): block weight: 5059 txs: 3 fees: 1063 sigops 405
2024-08-06T00:48:53Z [sv2] Send 0x71 NewTemplate id=1132 to client id=1
2024-08-06T00:48:53Z [sv2] Send 0x72 SetNewPrevHash to client id=1
2024-08-06T00:48:53Z [cormorant] MarkUnusedAddresses: Detected a used keypool item at index 0, mark all keypool items up to this item as used
2024-08-06T00:48:53Z CreateNewBlock(): block weight: 5059 txs: 3 fees: 1063 sigops 405
2024-08-06T00:48:53Z [sv2] Send 0x71 NewTemplate id=1133 to client id=2
2024-08-06T00:48:53Z [sv2] Send 0x72 SetNewPrevHash to client id=2
2024-08-06T00:48:53Z [cormorant] AddToWallet 2936ea56ed56d5e561de4f05ad8da99dba5f52cbcc905ec3452e68704fb3519e  new Confirmed (block=000000000000005aad1da1c2646a65e34c94089d7a51a9fe1ae3118169f18c82, height=38172, index=0)
2024-08-06T00:48:53Z [sv2] Received 0x73 RequestTransactionData from client id=2
2024-08-06T00:48:53Z [sv2] Send 0x74 RequestTransactionData.Success to client id=2

2024-08-06T06:22:07Z Bitcoin Core version v27.99.0-g44c8fa05bb16e1f7a00c8787703ef183a4b7cf51 (release build)
2024-08-06T06:22:07Z Script verification uses 3 additional threads
2024-08-06T06:22:07Z Using the 'sse4(1way),sse41(4way),avx2(8way)' SHA256 implementation
2024-08-06T06:22:07Z Using RdSeed as an additional entropy source
2024-08-06T06:22:07Z Using RdRand as an additional entropy source

### How did you obtain Bitcoin Core

Pre-built binaries

### What version of Bitcoin Core are you using?

sv2-tp-0.1.4 => v27.99.0-g44c8fa05bb16e1f7a00c8787703ef183a4b7cf51

### Operating system and version

Ubuntu 23.10

**Comment by Sjors** [comment]
Can you add `-loglevel=sv2:trace` to the configuration? That'll give me a more precise view of the last thing that happened. Also `-logtimemicros` will give you more accurate timestamps. 

In your log I'm seeing `RequestTransactionData.Success` as the last thing before the crash. But it's not clear if the crash happened there or a bit later.

**Comment by Georges760** [comment]
Yes I can and I will, but reproducing takes me multiple days... Hitting a block even on T4 is not immediate;)

**Comment by Georges760** [comment]
I can confirm with `sv2-tp-0.1.9` , I mined a block this night without crashing, so looking at the changelog, you have fixed this issue in 0.1.9.

**Comment by Sjors** [comment]
Thanks for testing and contrast on the testnet4 block!
